PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx

```diff
@@ -1734,172 +1786,177 @@ struct RobustFluctuationObservables {
-
-        double dauPosTPCcutOk = false;
-        if ((ptDauPos > ptTPCsectorCut && (fmod(phiDauPos + constPhiShift, TMath::Pi() / 9) > (MFsign < 0 ? fPhiCutExpPosHigh : fPhiCutExpNegHigh)->Eval(ptDauPos) || fmod(phiDauPos + constPhiShift, TMath::Pi() / 9) < (MFsign < 0 ? fPhiCutExpPosLow : fPhiCutExpNegLow)->Eval(ptDauPos))) || ptDauPos <= ptTPCsectorCut)
-          dauPosTPCcutOk = true;
-
-        if (dauPosTPCcutOk)
-          histosK0S.fill(HIST("posDau_phi_vs_pt_afterCut"), ptDauPos, phiDauPos);
-
-        if (ptDauPos > 0.5 && ptDauPos < 1.0)
-          histosK0S.fill(HIST("posDau_phi_pT_05_10"), phiDauPos);
-        else if (ptDauPos > 1.0 && ptDauPos < 2.0)
-          histosK0S.fill(HIST("posDau_phi_pT_10_20"), phiDauPos);
-        else if (ptDauPos > 2.0 && ptDauPos < 10.0)
-          histosK0S.fill(HIST("posDau_phi_pT_20_100"), phiDauPos);
-
-        // neg
-        histosK0S.fill(HIST("negDau_pt"), ptDauNeg);
-        histosK0S.fill(HIST("negDau_phi"), phiDauNeg);
-        histosK0S.fill(HIST("negDau_eta"), etaDauNeg);
-        histosK0S.fill(HIST("negDau_phi_vs_pt"), ptDauNeg, phiDauNeg);
-
-        double dauNegTPCcutOk = false;
-        if ((ptDauNeg > ptTPCsectorCut && (fmod(phiDauPos + constPhiShift, TMath::Pi() / 9) > (MFsign < 0 ? fPhiCutExpNegHigh : fPhiCutExpPosHigh)->Eval(ptDauNeg) || fmod(phiDauPos + constPhiShift, TMath::Pi() / 9) < (MFsign < 0 ? fPhiCutExpNegLow : fPhiCutExpPosLow)->Eval(ptDauNeg))) || ptDauNeg <= ptTPCsectorCut)
-          dauNegTPCcutOk = true;
-
-        if (dauNegTPCcutOk)
-          histosK0S.fill(HIST("negDau_phi_vs_pt_afterCut"), ptDauNeg, phiDauPos);
-
-        if (ptDauPos > 0.5 && ptDauPos < 1.0)
-          histosK0S.fill(HIST("negDau_phi_pT_05_10"), phiDauNeg);
-        else if (ptDauPos > 1.0 && ptDauPos < 2.0)
-          histosK0S.fill(HIST("negDau_phi_pT_10_20"), phiDauNeg);
-        else if (ptDauPos > 2.0 && ptDauPos < 10.0)
-          histosK0S.fill(HIST("negDau_phi_pT_20_100"), phiDauNeg);
-
-        // accept if both daughters are not at the TPC boundaries
-        if (dauPosTPCcutOk && dauNegTPCcutOk)
-          histosK0S.fill(HIST("hMassK0S_cut_TPC_boundaries_for_Dau"), v0.pt(), v0.mK0Short());
-      }
+        if (TMath::Abs(posDaughterTrack.tpcNSigmaPi()) > 3) // NSigmaTPCPion)
+          continue;
+        if (TMath::Abs(negDaughterTrack.tpcNSigmaPi()) > 3) // NSigmaTPCPion)
+          continue;
 
-      if (etaDauPos > 0.1 && etaDauPos<0.8 & etaDauNeg> 0.1 && etaDauNeg < 0.8)
-        histosK0S.fill(HIST("hMassK0ShortAfterSelectionVsPtEta01_08"), v0.pt(), v0.mK0Short());
-      if (etaDauPos > -0.8 && etaDauPos<-0.1 & etaDauNeg> - 0.8 && etaDauNeg < -0.1)
-        histosK0S.fill(HIST("hMassK0ShortAfterSelectionVsPtEta08_01"), v0.pt(), v0.mK0Short());
+        double ptDauPos = posDaughterTrack.pt();
+        double phiDauPos = posDaughterTrack.phi();
+        double ptDauNeg = negDaughterTrack.pt();
+        double phiDauNeg = negDaughterTrack.phi();
```

Comment:
Same comment here

Reply:
Hi Victor, thanks for the suggestion, I'll implement this at the next iteration of the code. 
(anyhow, currently, I'm focused on the BC-cut studies, so the K0S part is switched off by a flag)